mmcv/utils/config.py

```diff
@@ -555,20 +555,29 @@ def __setstate__(self, state):
         super(Config, self).__setattr__('_text', _text)
 
     def dump(self, file=None):
+        """
+            If a file argument is given, saves the config to that file using the format defined by the file argument extension.
+            Otherwise, returns a string representing the config.
+            The formatting of this returned string is defined by the extension of self.filename.
+            If self.filename is not defined, returns a string representation of a dict (lowercased and using ' for strings).
+        """
+        import mmcv
         cfg_dict = super(Config, self).__getattribute__('_cfg_dict').to_dict()
-        if self.filename.endswith('.py'):
-            if file is None:
+        if file is None:
+            if self.filename is None:
+                return mmcv.dump(cfg_dict)
```

Comment:
`mmcv.dump` will raise a `ValueError` if `file_format` and `file` is None at the same time. May we can directly dump a `.py` config file.